Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerGroupStore.cs

```diff
@@ -25,27 +25,37 @@ public SqlServerGroupStore(IAuthorizationDbContext authorizationDbContext)
 
         public async Task<Group> Add(Group model)
         {
-            var groupEntity = model.ToEntity();
+            var alreadyExists = await Exists(model.Name);
+            if (alreadyExists)
+            {
+                throw new AlreadyExistsException<Group>($"Group {model.Name} already exists. Please use a different GroupName.");
+            }
 
+            /*if (!string.IsNullOrEmpty(model.Id))
+            {
+                throw new BadRequestException<Group>("Id is generated internally and cannot be supplied by the caller.");
+            }*/
+
+            model.Id = Guid.NewGuid().ToString();
+            var groupEntity = model.ToEntity();
             _authorizationDbContext.Groups.Add(groupEntity);
             await _authorizationDbContext.SaveChangesAsync();
-
-            return model;
+            return groupEntity.ToModel();
         }
 
-        public async Task<Group> Get(string id)
+        public async Task<Group> Get(string name)
         {
             var group = await _authorizationDbContext.Groups
-                .Include(g => g.GroupRoles.Where(gr => !gr.IsDeleted))
+                .Include(g => g.GroupRoles)
                 .ThenInclude(gr => gr.Role)
-                .Include(g => g.GroupUsers.Where(gu => !gu.IsDeleted))
+                .Include(g => g.GroupUsers)
                 .ThenInclude(gu => gu.User)
-                .SingleOrDefaultAsync(g => g.Name.Equals(id, StringComparison.OrdinalIgnoreCase)
-                && !g.IsDeleted);
+                .SingleOrDefaultAsync(g => g.Name.Equals(name, StringComparison.OrdinalIgnoreCase)
+                                           && !g.IsDeleted);
 
             if (group == null)
             {
-                throw new NotFoundException<Group>($"Could not find {typeof(Group).Name} entity with ID {id}");
+                throw new NotFoundException<Group>($"Could not find {typeof(Group).Name} entity with ID {name}");
             }
 
             return group.ToModel();
```

Comment:
Since we removed the filter in IsDeleted above, do we need to filter on IsDeleted after execution of the query?

Reply:
Yes.